FCRM-6363 Encode polygon query strings

MickStein opened the pull request:
https://eaflood.atlassian.net/browse/FCRM-6363

Add encoder and decoder to polygons parsed through query params.

Commit:
Encode polygon query strings

https://eaflood.atlassian.net/browse/FCRM-6363

Add encoder and decoder to polygons parsed through query params.

Commit:
Added encoder function to frontend JS on map page button and decoder to backend router.

Commit:
Moved param decoder into separate module.

Commit:
Added testds and moved polygon decoding functions into utils file.

MickStein committed:
Merge branch 'development' into FCRM-6363

Commit:
Removed condole logs.

Commit:
Removed comments and installed mapbox package.

Commit:
Uninstall mapbox polyline as did not fix issue.

Commit:
Revert "Removed condole logs."

This reverts commit 083fcf56d646f5e8b234377866edaa3202aef287.

Commit:
Had to reinstall mapbox polyline as it was not recognised in node modules folder.

Commit:
Removed comments.

markfee reviewed (changes requested):
1 minor: test name change, and a major is the rename of the variable passed to the view.

Commit:
Made suggested changes and fixed sonar cloud issues.

MickStein commented:
This will stay as the main branch for the subtask changes to merge to:
https://eaflood.atlassian.net/browse/FCRM-6366
https://eaflood.atlassian.net/browse/FCRM-6365

MickStein committed:
Merge branch 'development' into FCRM-6363

MickStein committed:
Merge branch 'development' into FCRM-6363

MickStein committed:
FCRM-6365 Encode polygon URL contact page (#660)

* FCRM-6365 Encode polygon URL contact page

https://eaflood.atlassian.net/browse/FCRM-6365

Polygon in URL of contact page needs to be encoded to avoid excessive length URL bug. Merged into main feature branch.

MickStein committed:
Merge branch 'development' into FCRM-6363

MickStein committed:
Added tests to shape-util, unable to cover one line but it is covered. Unsure why.

markfee committed:
FCRM-6363 refactored checkParamsForPolygon to accept an object and fixed confirmation.js

MickStein commented on server/services/shape-utils.js:
```diff
@@ -118,14 +118,17 @@ const encodePolygon = (polygonArray) => {
+  return encodePolygon(queryObject.polygon)
+}
+
+// call like this: checkParamsForPolygon({...request.query, encode: false}) if you don't require an encodedPolygon
```
Does this comment need to be removed?

markfee replied:
I think keep it - it's a useful reminder to pass that in if we don't need the small overhead of an extra covnversion.
